### DISCUSSION TITLE
Save datetime list python

### DISCUSSION THREAD

**Discussion by wnesbv** [sqlalchemy/sqlalchemy#8205] [Usage Questions]
`

    time_start = sa.Column(sa.Date, nullable=True)

    time_end = sa.Column(sa.Date, nullable=True)

    reserve_period = sa.Column(sa.Date, nullable=True)
`
`

        if request.method == "POST":

            #...
            form = await request.form()
            start = form["time_start"]
            end = form["time_end"]
            #...
            time_start = datetime.strptime(
                start, settings.DATETIME
            )
            time_end = datetime.strptime(
                end, settings.DATETIME
            )

            generated = [
                time_start + timedelta(days=x)
                for x in range(0, (time_end - time_start).days + 1)
            ]
            reserve_period = []
            for period in generated:
                reserve_period.append(period))

            new_rsv = ReserveTimeFor()
            new_rsv.time_end = time_end
            new_rsv.time_start = time_start
            new_rsv.reserve_period = reserve_period

            session.add(new_rsv)
            session.refresh(new_rsv)
            await session.commit()

`

    raise TypeError(
sqlalchemy.exc.StatementError: (builtins.TypeError) SQLite Date type only accepts Python date objects as input.
[SQL: INSERT INTO reserve_time_for (description, reserve_time, time_start, time_end, reserve_period, owner_reserve, reserve_something) VALUES (?, ?, ?, ?, ?, ?, ?)]
[parameters: [{'time_end': datetime.datetime(2022, 7, 20, 0, 0), 'time_start': datetime.datetime(2022, 7, 5, 0, 0), 'owner_reserve': 1, 'reserve_period': ['2022-07 ... (134 characters truncated) ... 022-07-15', '2022-07-16', '2022-07-17', '2022-07-18', '2022-07-19', '2022-07-20'], 'reserve_something': 1, 'reserve_time': None, 'description': None}]]

**Comment by CaselIT** [comment]
Hi,

The date sql type can save a single, not a list. depending on the db you may use an Array sql type

**Reply by wnesbv** [reply]
about "The sql date type can save one, not a list." - I know..

**Reply by lelit** [reply]
So, what's the question exactly?
